lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3InputStream.java

```diff
@@ -190,11 +192,17 @@ private void seekStream()
         closeStream();
 
         try {
-            GetObjectRequest rangeRequest = request;
+            GetObjectRequest.Builder builder = request.toBuilder();
             if (nextReadPosition != 0) {
                 String range = "bytes=%s-".formatted(nextReadPosition);
-                rangeRequest = request.toBuilder().range(range).build();
+                builder.range(range);
             }
+            if (context.sseType() == S3FileSystemConfig.S3SseType.CUSTOMER) {
```

Comment:
can we reuse `S3Input::addEncryptionSettings`? Same below

Reply:
addressed